### DISCUSSION TITLE
Nested Module Slugs

### DISCUSSION THREAD

**Discussion by freshleafmedia** [area17/twill#1252] [Q&A]
I am looking to build a module which allows a tree of pages to be created like this:

- Home
- Product A
  - About
  - Docs
- Product B
  - About
  - Docs
- Contact

This is achievable using the current [Nested Module](https://twill.io/docs/#nested-module) setup but the problem is that I want the slugs to follow the same tree format:

```
/home
/product-a
/product-a/about
/product-a/docs
/product-b
/product-b/about
/product-b/docs
/contact
```

This doesn't seem to be possible currently because the [code that checks for slug uniqueness](https://github.com/area17/twill/blob/84eb34cfc747f213d45eaea487679ae385cf1c15/src/Models/Behaviors/HasSlug.php#L150-L171) doesn't take parent nodes into consideration.

Is there something I'm missing to make this work?

**Comment by pboivin** [comment]
Hi @freshleafmedia. You're right, I don't think this will be possible without customizing `HasSlug` to allow duplicates. As far as I can tell, there's no other constraint at the DB level on the slug uniqueness.

On the topic of nested items, a few helpers will be available in `2.6` to generate nested slugs and ancestor slugs :
https://github.com/area17/twill/blob/2.x/docs/.sections/crud-modules.md#working-with-self-nested-items
